lib/mixlib/shellout/windows/core_ext.rb

```diff
@@ -308,25 +309,66 @@ def create(args)
           end
 
           unless bool
-            raise SystemCallError.new("CreateProcessAsUserW (You must hold the 'Replace a process level token' permission)", FFI.errno)
+            raise SystemCallError.new("CreateProcessAsUserW (You must hold the 'Replace a process level token' permission. Restart your machine after adding this right to make it effective.)", FFI.errno)
           end
         else
-          bool = CreateProcessWithLogonW(
-            logon,                  # User
-            domain,                 # Domain
-            passwd,                 # Password
-            LOGON_WITH_PROFILE,     # Logon flags
-            app,                    # App name
-            cmd,                    # Command line
-            hash["creation_flags"], # Creation flags
-            env,                    # Environment
-            cwd,                    # Working directory
-            startinfo,              # Startup Info
-            procinfo                # Process Info
-          )
+          if hash["elevated"]
+            token = FFI::MemoryPointer.new(:ulong)
+
+            bool = LogonUserW(
+              logon,                      # User
+              domain,                     # Domain
+              passwd,                     # Password
+              LOGON32_LOGON_BATCH,        # Logon Type
+              LOGON32_PROVIDER_DEFAULT,   # Logon Provider
+              token                       # User token handle
+            )
 
-          unless bool
-            raise SystemCallError.new("CreateProcessWithLogonW", FFI.errno)
+            unless bool
+              raise SystemCallError.new("LogonUserW", FFI.errno)
+            end
+
+            token = token.read_ulong
+
+            begin
+              bool = CreateProcessAsUserW(
+                token,                  # User token handle
+                app,                    # App name
+                cmd,                    # Command line
+                process_security,       # Process attributes
+                thread_security,        # Thread attributes
+                inherit,                # Inherit handles
+                hash["creation_flags"], # Creation Flags
+                env,                    # Environment
+                cwd,                    # Working directory
+                startinfo,              # Startup Info
+                procinfo                # Process Info
+              )
+            ensure
+              CloseHandle(token)
+            end
+
+            unless bool
+              raise SystemCallError.new("CreateProcessAsUserW (You must hold the 'Replace a process level token' permission. Restart your machine after adding this right to make it effective.)", FFI.errno)
```

Comment:
There are several error possibilities here. We should only suggest adding the user permission if the error is `ERROR_PRIVILEGE_NOT_HELD ` and otherwise return the formatted message from the error reported by `GetLastError`.